### DISCUSSION TITLE
Something broke ... getting error 500

### DISCUSSION THREAD

**Discussion by dzg** [YOURLS/YOURLS#3767] [Q&A] [answered]
Everything was working fine, but now, if I try to add a URL, it just hangs (with spinner gif) and I see a network call to `https://mysite.com/admin/admin-ajax.php?action=...` returning `500 Internal Server Error`

I tried an API call, like `https://mysite.com/yourls-api.php?signature=xxxxxxx&action=shorturl&url=https://example.com&title=title&format=json` and also got a 500

But if I just do `https://mysite.com/yourls-api.php?signature=xxxxxxx&action=shorturl&format=json` I get 
```
{
status: "fail",
code: "error:nourl",
message: "Missing or malformed URL",
errorCode: "400",
statusCode: "400"
}
```
... so something is listening.

(Adding `define('YOURLS_DEBUG', true);` doesn't show anything upon trying to add a URL)

**Comment by dgw** [comment]
We're not going to be able to tell you much without logs. There should be a PHP error log on your system, unless your hosting provider has disabled it for some reason.

Hanging on the spinning icon when shortening a new link is one of the most frequent issues, and has a dedicated troubleshooting section in YOURLS' documentation: https://yourls.org/docs/guide/troubleshooting/frequent-issues#endless-spinning-icon

**Reply by dzg** [reply]
I turned on PHP errors and seeing a lot of `Notice: yourls_string2htmlid is deprecated since version 1.8.3! Use yourls_unique_element_id instead. in /home4/user/public_html/site.com/includes/functions.php on line 1227`

Also `error_log` shows something like this for every hang:
```
[01-Jun-2024 00:29:19 UTC] PHP Fatal error:  Uncaught TypeError: in_array(): Argument #2 ($haystack) must be of type array, null given in /home4/user/public_html/site.com/includes/functions-shorturls.php:221
Stack trace:
#0 /home4/user/public_html/site.com/includes/functions-shorturls.php(221): in_array('3', NULL)
#1 /home4/user/public_html/site.com/includes/functions-shorturls.php(414): yourls_keyword_is_reserved('3')
#2 /home4/user/public_html/site.com/includes/functions-shorturls.php(128): yourls_keyword_is_free('3')
#3 /home4/user/public_html/site.com/includes/functions-api.php(21): yourls_add_new_link('aaa', '3', 'aaa')
#4 /home4/user/public_html/site.com/includes/functions-plugins.php(221): yourls_api_action_shorturl(false)
#5 /home4/user/public_html/site.com/yourls-api.php(35): yourls_apply_filter('api_action_shor...', false)
#6 {main}
  thrown in /home4/user/public_html/site.com/includes/functions-shorturls.php on line 221
```

**Reply by dzg** [reply]
Actually the stack trace seems to happen only when trying to add URL via API.
Adding via webpage does not add anything to log.

**Reply by dzg** [reply]
Also this is my `.htaccess`

```
# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /yourls-loader.php [L]
</IfModule>
# END YOURLS

# BEGIN cPanel-generated php ini directives, do not edit
# Manual editing of this file may result in unexpected behavior.
# To make changes to this file, use the cPanel MultiPHP INI Editor (Home >> Software >> MultiPHP INI Editor)
# For more information, read our documentation (https://go.cpanel.net/EA4ModifyINI)
<IfModule php8_module>
   php_flag display_errors Off
   php_value max_execution_time 60
   php_value max_input_time 60
   php_value max_input_vars 1000
   php_value memory_limit 512M
   php_value post_max_size 516M
   php_value session.gc_maxlifetime 1440
   php_value session.save_path "/var/cpanel/php/sessions/ea-php82"
   php_value upload_max_filesize 512M
   php_flag zlib.output_compression Off
</IfModule>
<IfModule lsapi_module>
   php_flag display_errors Off
   php_value max_execution_time 60
   php_value max_input_time 60
   php_value max_input_vars 1000
   php_value memory_limit 512M
   php_value post_max_size 516M
   php_value session.gc_maxlifetime 1440
   php_value session.save_path "/var/cpanel/php/sessions/ea-php82"
   php_value upload_max_filesize 512M
   php_flag zlib.output_compression Off
</IfModule>
# END cPanel-generated php ini directives, do not edit
```

**Reply by dgw** [reply, marked as answer]
Do you have `$yourls_reserved_URL` defined in your `config.php`? The stack trace you were able to post points to that variable being NULL, as if it is never defined in your config file.

Minimal code example that throws the same error:

```php
<?php

function some( $thing ) {
	global $yourls_reserved_URL;
	
	return in_array( $thing, $yourls_reserved_URL );
}

print(some("3"));
```

[Interactive demo](https://onlinephp.io?s=s7EvyCjg5eLlSivNSy7JzM9TKM7PTdVQUCnJyMxLV9BUqObl4kzPyU9KzFFQqcwvLcopji9KLU4tKktNiQ8N8rEGSgNRUWpJaVGeQmZefGJRUWIlTL8OVj0KmkBdtSBLC4oy80o0wDYqGStpaloDAA%2C%2C&v=8.2.18)

**Reply by dzg** [reply]
Ah, I see. It was because I deleted the `$yourls_reserved_URL` from `config.php` because I don't really care about that. Didn't realize it was required.

Thank you!
